### PR TITLE
fixes #8941 - dynflow refresh subscriptions (auto-attach)

### PR DESCRIPTION
**Pull request by thomasmckay**

### CODE REVIEW

File: app/lib/actions/katello/system/update.rb [hunk 1]
```diff
@@ -23,8 +23,8 @@ def plan(system, sys_params)
           plan_action(::Actions::Katello::Foreman::HostUpdate, system)
           plan_action(::Actions::Pulp::Consumer::Update, system) if ::Katello.config.use_pulp
           plan_action(::Actions::Candlepin::Consumer::Update, system) if ::Katello.config.use_cp
-          plan_action(ElasticSearch::Reindex, system) if ::Katello.config.use_elasticsearch
           plan_action(::Actions::Candlepin::Consumer::RefreshSubscriptions, system, sys_params) if ::Katello.config.use_cp
+          plan_action(ElasticSearch::Reindex, system) if ::Katello.config.use_elasticsearch
```
Comment (thomasmckay) [inline app/lib/actions/katello/system/update.rb]: Moved elasticsearch after subscription refresh so candlepin data will be reflected properly

File: app/lib/actions/katello/system/refresh_subscriptions.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+module Actions
+  module Katello
+    module System
+      class RefreshSubscriptions < Actions::EntryAction
+        middleware.use ::Actions::Middleware::RemoteAction
+
+        def plan(system)
+          system.disable_auto_reindex!
+          action_subject system
+          plan_action(::Actions::Candlepin::Consumer::RefreshSubscriptions, system, :autoheal => true) if ::Katello.config.use_cp
```
Comment (jlsherrill) [inline app/lib/actions/katello/system/refresh_subscriptions.rb]: autoheal is hardcoded to true here

**Comment by thomasmckay** [comment]
Talking to @jlsherrill and concluded that the RefreshSubscriptions should be renamed to AutoAttach. The API can maintain old route (w/ deprecated flag?) but should be changed to auto_attach as well.

Things to confirm works: 
1. toggle auto-attach flag to true on content host triggers a call
2. auto-attach UI/CLI command triggers a call (does not alter flag on content host)
3. auto-attach on all content hosts in org triggers call to all (does not alter flag on individual content hosts)

**Comment by jlsherrill** [comment]
@thomasmckay the /systems api is all deprecated anyways, so i think we are okay to change?

I would add:
4.  RefreshSubscription actions should be renamed to AutoAttach
5.  if test currently in Candlepin:RefreshSubscriptions action should be moved to Katello::System::Update

**Comment by thomasmckay** [comment]
1. CONFIRM toggle auto-attach flag to true on content host triggers a call
2. CONFIRM UI (no CLI) auto-attach UI/CLI command triggers a call (does not alter flag on content host)
3. CONFIRM auto-attach on all content hosts in org triggers call to all (does not alter flag on individual content hosts)
4. DONE RefreshSubscription actions should be renamed to AutoAttach
5. DONE if test currently in Candlepin:RefreshSubscriptions action should be moved to Katello::System::Update

**Comment by jlsherrill** [comment]
ACK

**Pull request merged**